FWCore/Framework/src/WorkerMaker.cc

```diff
@@ -58,6 +58,11 @@ namespace edm {
       MakeModuleParams const& p,
       signalslot::Signal<void(ModuleDescription const&)>& pre,
       signalslot::Signal<void(ModuleDescription const&)>& post) const {
+    // Add process_name for SwitchProducer
+    if (p.pset_->getParameter<std::string>("@module_type") == "SwitchProducer") {
```

Comment:
Would doing
```C++
p.pset_->getParameter<std::string>("@module_type"s) == "SwitchProducer"s) {
```
avoid the need to create and destroy the strings?

Reply:
I just played around with that idea on godbolt.org and it did not make a difference.

Reply:
In principle making the `"@module_type"` to a constant could help (an `std::string` gets created eventtually when a `char const*` is passed) (*), the `"SwitchProducer"` I'd expect to stay as a `char[]` constant and lead to picking `operator==(std::string const&, char const*)` overload.

(*) but then I'd change all such strings in this file, so maybe for a subsequent PR.